src/nnet3/nnet-simple-component.cc

```diff
@@ -3778,10 +3828,14 @@ void ConvolutionComponent::Write(std::ostream &os, bool binary) const {
   WriteBasicType(os, binary, filt_x_dim_);
   WriteToken(os, binary, "<FiltYDim>");
   WriteBasicType(os, binary, filt_y_dim_);
+  WriteToken(os, binary, "<FiltZDim>");
+  WriteBasicType(os, binary, filt_z_dim_);
   WriteToken(os, binary, "<FiltXStep>");
   WriteBasicType(os, binary, filt_x_step_);
   WriteToken(os, binary, "<FiltYStep>");
   WriteBasicType(os, binary, filt_y_step_);
+  WriteToken(os, binary, "<FiltZStep>");
```

Comment:
This breaks compatibility with CNNs created using old binaries. You should make it clear in the commit message that the CNN models are no longer accessible using this new code.